docs/books/user-guide/routing.adoc

```diff
@@ -102,6 +102,17 @@ Link routing supports local transactions to a broker.
 +
 With a link route, consumers can provide server-side selectors for broker subscriptions.
 
+[id='considerations-for-broker-connection-failure-handling']
+=== Considerations for Broker Connection Failure Handling
+
+The message routing and link routing mechanisms both enable you to transfer messages to brokers. However, they differ in how they behave if the connection to the broker is lost. If you require the ability for the connection to be reestablished automatically, you should use message routing (with autolinks).
+
+With link routing::
+The client appears to be directly connected to the broker even though it is actually connected to a router, which is link routed to the broker. If the broker goes offline, the router sees the connection failure and closes all of the routed links that were attached to the broker. The client's connection to the router, however, remains open. Therefore, the client does not see the connection failure between the broker and the router and does not attempt to reestablish it.
```

Comment:
I think this needs to be more explicit about what is routed and what is not.  When a broker with link-routes goes offline, the routed links are detached.  The router will continue to attempt to reconnect to the broker (or its backup) at which point, the link-routes will be reactivated (i.e. they will become reachable again).

The client does not see a dropped connection, but it does see the detached links (senders and/or receivers).  The client has the option of attempting to re-attach lost links.  The reattach will either go to a different broker, fail due to no available destination, or be reattached to the original broker when it comes back online.